src/careamics/lightning/lightning_module.py

```diff
@@ -202,6 +205,201 @@ def configure_optimizers(self) -> Any:
+    lr_scheduler_name : str
+        Learning rate scheduler name.
+    """
+
+    def __init__(self, algorithm_config: Union[AlgorithmConfig, dict]) -> None:
+        """Lightning module for CAREamics.
+
+        This class encapsulates the a PyTorch model along with the training, validation,
+        and testing logic. It is configured using an `AlgorithmModel` Pydantic class.
+
+        Parameters
+        ----------
+        self.algorithm_config : Union[AlgorithmModel, dict]
+            Algorithm configuration.
+        """
+        super().__init__()
+        # if loading from a checkpoint, AlgorithmModel needs to be instantiated
+        self.algorithm_config = (
+            AlgorithmConfig(**algorithm_config)
+            if isinstance(algorithm_config, dict)
+            else algorithm_config
+        )
+
+        # create model and loss function
+        self.model: nn.Module = model_factory(self.algorithm_config.model)
+        self.likelihood = likelihood_factory(self.algorithm_config.likelihood)
+        self.noise_model = noise_model_factory(self.algorithm_config.noise_model)
+        self.loss_parameters = LVAELossParameters
+        # TODO how to modify these ?
+        self.loss_func = loss_factory(self.algorithm_config.loss)
+
+        # save optimizer and lr_scheduler names and parameters
+        self.optimizer_name = self.algorithm_config.optimizer.name
+        self.optimizer_params = self.algorithm_config.optimizer.parameters
+        self.lr_scheduler_name = self.algorithm_config.lr_scheduler.name
+        self.lr_scheduler_params = self.algorithm_config.lr_scheduler.parameters
+
+    def forward(self, x: Any) -> Any:
+        """Forward pass.
+
+        Parameters
+        ----------
+        x : Any
+            Input tensor.
+
+        Returns
+        -------
+        Any
+            Output tensor.
+        """
+        return self.model(x)  # TODO Different model can have more than one output
+
+    def training_step(self, batch: Tensor, batch_idx: Any) -> Any:
+        """Training step.
+
+        Parameters
+        ----------
+        batch : Tensor
+            Input batch.
+        batch_idx : Any
+            Batch index.
+
+        Returns
+        -------
+        Any
+            Loss value.
+        """
+        x, *aux = batch
+        out = self.model(x)
+        target = aux[0]
+        # TODO rethink loss parameters
+        self.loss_parameters.current_epoch = self.current_epoch
+        self.loss_parameters.inputs = x
+        self.loss_parameters.mask = ~((target == 0).reshape(len(target), -1).all(dim=1))
+        self.loss_parameters.likelihood = self.likelihood
+        self.loss_parameters.noise_model = self.noise_model
+        loss = self.loss_func(out, target, self.loss_parameters)  # TODO ugly ?
+
+        self.log_dict(loss, on_step=True, on_epoch=True, prog_bar=True, logger=True)
+        return loss
+
+    def validation_step(self, batch: Tensor, batch_idx: Any) -> None:
```

Comment:
Seems like a lot has been removed from the validation step in src/careamics/lvae_training/lightning_module.py is everything in there definitely useless?

Reply:
It wasn't added, not "was removed"